tests/cephfs/cephfs_nfs/test_cephfs_nfs_ha_node_replacement.py

```diff
@@ -0,0 +1,367 @@
+        else:
+            print("Object not found in interm_list")
+        active_nfs_after = get_active_nfs(interm_list, virtual_ip)
+        for i in interm_list:
+            fs_util_v1.wait_for_service_to_be_in_running(client1, i)
+        haproxy_ls_after = fs_util_v1.get_deamon_status(client1, "haproxy")
+        keepalived_ls_after = fs_util_v1.get_deamon_status(client1, "keepalived")
+        haproxy_ls_after_hostnames = [entry["hostname"] for entry in haproxy_ls_after]
+        keepalived_ls_after_hostnames = [
+            entry["hostname"] for entry in keepalived_ls_after
+        ]
+        if active_nfs.node.hostname == active_nfs_after.node.hostname:
+            raise CommandFailed(
+                f"The active node did not change even after power off."
+                f"\nPrevious active node: {active_nfs.node.hostname}"
+                f"\nCurrent active node: {active_nfs_after.node.hostname}"
+            )
+        if (
+            haproxy_ls_after_hostnames == haproxy_ls_hostnames
+            or keepalived_ls_after_hostnames == keepalived_ls_hostnames
+        ):
+            raise CommandFailed(
+                f"The ha proxy node did not change even after power off ."
+                f"\nPrevious haproxy nodes: {haproxy_ls_hostnames}"
+                f"\nCurrent haproxy nodes: {haproxy_ls_after_hostnames}"
+                f"\nPrevious keepalived nodes: {keepalived_ls_hostnames}"
+                f"\nCurrent keepalived nodes: {keepalived_ls_after_hostnames}"
+            )
+        log.info(
+            f"The active node changed after Power off."
+            f"\nPrevious active node: {active_nfs.node.hostname}"
+            f"\nCurrent active node: {active_nfs_after.node.hostname}"
+        )
+        orch_formatted_string = " ".join(
+            [f"{node.node.hostname}" for node in interm_list]
+        )
+        log.info("Update with New node")
+        nfs_fstr = """
+service_type: ingress
+service_id: nfs.cephnfs
+placement:
+  count: 2
+  hosts:
+{hosts}
+spec:
+  backend_service: nfs.cephnfs
+  frontend_port: 2049
+  monitor_port: 9000
+  virtual_ip: 10.8.128.100/21
+""".format(
+            hosts="".join([f"   - {node.node.hostname}\n" for node in interm_list])
+        )
+
+        filename = "nfs.yaml"
+        nfs_file = clients[0].remote_file(
+            sudo=True,
+            file_name=filename,
+            file_mode="w",
+        )
+        nfs_file.write(nfs_fstr)
+        nfs_file.flush()
+        client1.exec_command(sudo=True, cmd=f"ceph orch apply -i {filename}")
+        client1.exec_command(
+            sudo=True,
+            cmd=f'ceph orch apply nfs {nfs_name} --placement "2 {orch_formatted_string}"',
+        )
+
+        active_nfs_ipmi = IPMIPowerControl(
+            host=f"{backend_server[1].node.hostname}.ipmi.ceph.redhat.com",
+            username="inktank",
+            password="ApGNXcA7",
+        )
+        log.info(f"Powering Off {backend_server[1].node.hostname}")
```

Comment:
please check the logs. Current active nfs node - magna025, however using backend_server[1] -> fixed entry we are powering off other node magna026. If the intention is to power off current active node, then the variables used needs to be revisited.

Reply:
As per test cases , When we deploy nfs cluster we will have 2 active nfs servers deployed.
We are going to power off only those 2 servers which we have captured as part of backend_server.
with this we are going to power off the nodes which are active while cluster was created  
Before this power off we are reconfiguring the cluster by adding new node on that node we are checking the services have been migrated or not